### PR TITLE
feat(TimeInput): add component

### PR DESCRIPTION
**Pull request by LucidityDesign**

### CODE REVIEW

File: src/components/controls/inputs/TimeInput/useRangeTimeInput.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  onChangeTo: (event: ChangeEvent<HTMLInputElement>) => void
+  maxFrom?: string | undefined
+  minTo?: string | undefined
+  statusFrom: Status
+  statusTo: Status
+}
+
+export type Status = "fail" | "neutral"
+
+export const useRangeTimeInput = (
```
Comment (filiptammergard) [inline src/components/controls/inputs/TimeInput/useRangeTimeInput.ts:16]: Maybe some JSDocs on this hook as well? Explaining what this hooks does and what happens when range is invalid etc.

File: src/components/controls/inputs/TimeInput/TimeInput.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  },
+} satisfies Story
+
+const RangeTemplate = (): JSX.Element => {
+  const { valueFrom, onChangeFrom, valueTo, onChangeTo, minTo, maxFrom, statusFrom, statusTo } =
+    useRangeTimeInput()
+  return (
+    <HorizontalLayout>
+      <TimeInput
+        /* status={statusFrom} */
```
Comment (filiptammergard) [inline src/components/controls/inputs/TimeInput/TimeInput.stories.tsx:103]: Is this intended? Seems like `statusForm` is currently unused

**Review by filiptammergard — COMMENTED** [review]
Just thinking out load: Do we need some time format prop? See https://www.notion.so/einride/Dates-and-times-in-Saga-cd3a770ce2374869938f8dc3c6459f48?pvs=4

Mantine uses `format: "12" | "24"`, but maybe that's not possible with the native `<input type="time" />`?

Should it be possible to choose between `:` and `.` as separators?

**Comment by LucidityDesign** [comment]
> Just thinking out load: Do we need some time format prop? See https://www.notion.so/einride/Dates-and-times-in-Saga-cd3a770ce2374869938f8dc3c6459f48?pvs=4
> 
> Mantine uses `format: "12" | "24"`, but maybe that's not possible with the native `<input type="time" />`?
> 
> Should it be possible to choose between `:` and `.` as separators?

This is not possible with the native time input. It will choose 12/24h format based on the user preference

**Review by filiptammergard — APPROVED** [review]
Looks good! 👏

**Pull request merged**